app/mod_socketio/events.py

```diff
@@ -1,43 +1,48 @@
-import csv
-import time
-
-from flask import Blueprint
+from flask import Blueprint, jsonify
 from flask_socketio import emit
-import Adafruit_TCS34725
 
-from app import socketio, logger, app
-from app.mod_socketio.matching import closest_color
-from app.devices.led import RGBLed
+from app import socketio, logger, app, db
+from app.mod_colors.models import Color
+from app.devices import sensor
 
 mod_socketio = Blueprint('socketio', __name__, url_prefix='/')
 
-# Global state
-colors = None
-tcs = Adafruit_TCS34725.TCS34725(
-    integration_time=Adafruit_TCS34725.TCS34725_INTEGRATIONTIME_700MS)
 
-led = RGBLed((11, 13, 15))
+def get_all_colors():
+    colors=Color.query.all()
+    return [{'rgb': color.rgb, 'color': color.color} for color in colors]
 
-# representing state of the sensor LED
-light_on = True
 
-# read colors from csv
-with open('colors.csv') as colors_csv:
-    colors_reader = csv.reader(colors_csv)
-    colors = [tuple(row) for row in colors_reader]
+@socketio.on_error()
+def error_handler(e):
+    logger.exception(e)
 
 
-def set_light_state():
-    # switch the sensor LED
-    tcs.set_interrupt(not light_on)
+@socketio.on('select')
+def select(req):
+    emit('colors', get_all_colors())
 
 
-set_light_state()
+@socketio.on('insert')
+def insert(color):
+    db.session.add(Color(rgb=color['rgb'], color=color['color']))
+    db.session.commit()
+    emit('colors', get_all_colors())
 
 
-@socketio.on_error()
-def error_handler(e):
-    logger.exception(e)
+@socketio.on('delete')
+def delete(color):
+    color = Color.query.filter(Color.rgb == color['rgb']).first()
```

Comment:
Myślę że poprawnie byłoby przekazywać `id` koloru a nie `rgb` z bazy, gdyż dopuszczamy występowanie tego samego kolowu na liście, przy czym `first()` nie rozwiązuje problemu bo usuwany obiekt nie musi to być pierwszym z brzegu.
Najlepiej wg mnie byłoby mieć kolory (`Colors`) w bazie danych i osobną tablę z zapisanymi kolorami z polami `FK Colors.id ` oraz `PK id` i wtedy tylko usuwać po id wpisu a nie przekazywać `rgb` albo `name`.